frontend/src/metabase/static-viz/components/XYChart/utils/ticks.ts

```diff
@@ -151,3 +159,28 @@ export const getYTickWidths = (
         : 0,
   };
 };
+
+export function getBetterXTicks(
+  xScaleType: XAxisType,
+  xScale: XScale["scale"],
+  numTicks: number,
+) {
+  let minLengthTicks = getTicks(xScale, numTicks);
+  if (xScaleType === "timeseries") {
+    const candidateTickIntervals = [
+      timeWeek.every(2) as TimeInterval,
+      timeMonth.every(2) as TimeInterval,
+    ];
+    candidateTickIntervals
+      .map(tickInterval => getTicks(xScale, tickInterval as unknown as number))
+      .filter(ticks => ticks.length > 0)
+      .forEach(ticks => {
+        console.log(ticks);
```

Comment:
Don't forget to remove this console.log. Also, wdyt about coming up with a more descriptive function name?

Reply:
Haha, I was so in a hurry 🤦  Agree that the function could also be improved.